src/Features/CSharp/Portable/Completion/KeywordRecommenders/AssemblyKeywordRecommender.cs

```diff
@@ -17,15 +15,22 @@ public AssemblyKeywordRecommender()
 
         protected override bool IsValidContext(int position, CSharpSyntaxContext context, CancellationToken cancellationToken)
         {
-            if (context.IsTypeAttributeContext(cancellationToken))
+            var token = context.TargetToken;
+
+            if (token.Kind() != SyntaxKind.OpenBracketToken)
             {
-                var token = context.LeftToken;
-                var type = token.GetAncestor<MemberDeclarationSyntax>();
+                return false;
+            }
 
-                return type == null || type.IsParentKind(SyntaxKind.CompilationUnit);
+            if (token.Parent.Kind() == SyntaxKind.AttributeList)
+            {
+                var attributeList = token.Parent;
+                var previousSyntax = attributeList.Parent;
+                return previousSyntax is CompilationUnitSyntax || previousSyntax.Parent is CompilationUnitSyntax;
             }
 
-            return false;
+            var parentSyntax = token.Parent;
+            return parentSyntax is SkippedTokensTriviaSyntax;
```

Comment:
this part is not clear.  As with my other comment:

1. there's no explanation in the code as to why this code is here or why it's correct.
2. this looks like you'll show 'assembly' after any skipped `[` token.  That doesn't seem like something we'd want.